src/libraries/Microsoft.PowerFx.Interpreter/Values/QueryableTableValue.cs

```diff
@@ -1,23 +1,23 @@
 ﻿// Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
 
-using System;
 using System.Collections.Generic;
+using System.Runtime.CompilerServices;
 using System.Threading.Tasks;
 using Microsoft.PowerFx.Core.IR;
 
 namespace Microsoft.PowerFx.Types
 {
     public abstract class QueryableTableValue : TableValue
     {
-        private Lazy<Task<List<DValue<RecordValue>>>> _lazyTaskRows;
+        private IAsyncEnumerable<DValue<RecordValue>> _cachedRows;
```

Comment:
>_cachedRows [](http://example.com/codeflow?start=54&length=11)

Removing Lazy<List> will remove the cache. 
The enum is not actually caching.

Reply:
The test would be to verify we only enumerate once even when we call foreach many times: 

foreach(var row in Rows) { }
foreach(var row in Rows) { }
foreach(var row in Rows) { }

Reply:
I need to add a UT To validate this important point and ensure we call GetRowsAsync() only once in this scenario